utils/wasm-gen/src/tests.rs

```diff
@@ -157,6 +167,245 @@ fn injecting_addresses_works() {
+            "syscall: {}",
+            syscall.to_str()
+        );
+    }
+}
+
+#[derive(Clone)]
+struct MemoryWrite {
+    offset: u32,
+    content: Vec<u8>,
+}
+
+fn get_params_for_syscall_to_fail(
+    syscall: &SysCallName,
+) -> (SysCallsParamsConfig, Option<MemoryWrite>, Payload) {
+    match *syscall {
+        SysCallName::ReplyPush => {
+            let mut params_config = SysCallsParamsConfig::constant_value(i32::MAX as i64);
+            params_config.add_rule(ParamType::Ptr(None), SysCallParamAllowedValues::zero());
+
+            (params_config, None, Payload::default())
+        }
+        SysCallName::ReplyPushInput => {
+            let mut params_config = SysCallsParamsConfig::constant_value(i32::MAX as i64);
+            params_config.add_rule(ParamType::Ptr(None), SysCallParamAllowedValues::zero());
+
+            let mut payload = Payload::default();
+            payload.extend_with(0);
+
+            params_config.add_rule(
+                ParamType::Size,
+                // offset = length = MAX_PAYLOAD_SIZE / 2, so MAX_PAYLOAD_SIZE / 2 would be read
+                // when the payload with max possible length is provided.
+                SysCallParamAllowedValues::constant(payload.inner().len() as i64 / 2),
+            );
+
+            (params_config, None, payload)
+        }
+        SysCallName::Read => {
+            let mut params_config = SysCallsParamsConfig::constant_value(0);
+            params_config.add_rule(
+                ParamType::Size,
+                SysCallParamAllowedValues::constant(i32::MAX as i64),
+            );
+            (params_config, None, Payload::default())
+        }
+        SysCallName::PayProgramRent => (
+            SysCallsParamsConfig::constant_value(0),
+            Some(MemoryWrite {
+                offset: 0,
+                content: std::iter::repeat(255).take(128).collect(),
+            }),
+            Payload::default(),
+        ),
+        _ => (
+            SysCallsParamsConfig::constant_value(0),
+            None,
+            Payload::default(),
+        ),
+    }
+}
+
+fn execute_wasm_with_syscall_injected(
+    syscall: SysCallName,
+    ignore_fallible_errors: bool,
+    params_config: SysCallsParamsConfig,
+    initial_memory_write: Option<MemoryWrite>,
+    payload: Payload,
+) -> TerminationReason {
+    use gear_backend_common::{BackendReport, Environment};
+    use gear_backend_wasmi::WasmiEnvironment;
+    use gear_core::{
+        gas::{GasAllowanceCounter, GasCounter, ValueCounter},
+        memory::AllocationsContext,
+        message::{ContextSettings, DispatchKind, IncomingDispatch, MessageContext},
+        reservation::GasReserver,
+    };
+    use gear_core_processor::{configs::PageCosts, ProcessorContext, ProcessorExternalities};
+
+    const INITIAL_PAGES: u16 = 1;
+    const INJECTED_SYSCALLS: u32 = 8;
+
+    let buf = vec![0; UNSTRUCTURED_SIZE];
+    let mut unstructured = Unstructured::new(&buf);
```

Comment:
you create program from zero bytes?

Reply:
nope, from array filled with zeroes

Reply:
ha, i see, the test instantiates wasm with only one function that has only one sys-call. It's ok in this case